### DISCUSSION TITLE
Type errors "Cannot be used as a JSX component" after update from v18.2.1 [@types/react]

### DISCUSSION THREAD

**Discussion by jinsley8** [DefinitelyTyped/DefinitelyTyped#65447] [Issues with a @types package] [answered]
I updated `@types/react` a couple days ago from `v18.2.1` to `v18.2.6` and now I'm getting some weird type errors. Not sure if these are expected or if they are bugs.

Here is one example:
```js
'SearchResults' cannot be used as a JSX component.
  Its element type 'ReactElement<any, any> | Component<SearchBarResultsProps, any, any> | null' is not a valid JSX element.
    Type 'Component<SearchBarResultsProps, any, any>' is not assignable to type 'Element | ElementClass | null'.
      Type 'Component<SearchBarResultsProps, any, any>' is not assignable to type 'ElementClass'.
        The types returned by 'render()' are incompatible between these types.
          Type 'React.ReactNode' is not assignable to type 'import("/node_modules/@types/react-dom/node_modules/@types/react/ts5.0/index").ReactNode'.
```

I seems like this is happening most of the time with components that are modals or popovers. If this is expect then how would I resolve the type error?

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment, marked as answer]
`/node_modules/@types/react-dom/node_modules/@types/react/ts5.0/index` <- looks like you have two different versions of `@types/react`. Remain one.

**Reply by eps1lon** [reply]
That's what I think as well. https://solverfox.dev/writing/duplicate-types-react explains the cause and possible fixes in more detail

**Reply by jinsley8** [reply]
Thanks, I tried deleting `node_modules` and `yarn.lock` and re-installing but that didn't help.

When I run `yarn why @types/react` it shows the following:

<img width="524" alt="Screenshot 2023-05-09 at 11 57 51 AM" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/5669143/d2983387-96c6-4266-a0a3-17f75c93fbdd">

The dependency in these packages is:
```bash
"dependencies": {
        "@types/react": "*"
},
```
    
I also tried setting resolutions in package.json:
```bash
"resolutions": {
        "@types/react": "18.2.6"
},
```

The one error says: 
`Type 'React.ReactNode' is not assignable to type 'import("/node_modules/@types/react-dom/node_modules/@types/react/ts5.0/index").ReactNode'.`

It looks like there is only one version of `@types/react`:
<img width="599" alt="Screenshot 2023-05-09 at 12 26 27 PM" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/5669143/bd9ecb17-7ee5-4222-aed6-cf6acf322385">

**Reply by jinsley8** [reply]
I added a path to `tsconfig.json` but not sure if this is recommended or if it's just masking the problem but it gets rid of the error:

```bash
"paths": {
      "react": [ "./node_modules/@types/react" ],
},
```

I had tried adding resolutions in `package.json` as others suggest in other issues but that did not resolve the issue.

```bash
"resolutions": {
    "@types/react": "18.2.6",
    "@types/react-dom": "18.2.4"
},
```

**Reply by eps1lon** [reply]
> Type 'React.ReactNode' is not assignable to type 'import("/node_modules/@types/react-dom/node_modules/@types/react/ts5.0/index").ReactNode'.

This error says that you have multiple versions of `@types/react` installed. Without a cloneable, minimal repro we won't be able to help you.
